go/vt/vtgate/engine/ordered_aggregate.go

```diff
@@ -102,8 +102,9 @@ type AggregateParams struct {
 	WAssigned   bool
 	CollationID collations.ID
 
-	Alias string `json:",omitempty"`
-	Expr  sqlparser.Expr
+	Alias    string `json:",omitempty"`
+	Expr     sqlparser.Expr
+	Original *sqlparser.AliasedExpr
```

Comment:
Why do we need `original`? The only uses I see should be possible to handle using the Expr field, right?

Reply:
`Expr` field is the AST node that contains the aggregate function (i.e. `count(id)`), whereas `Original` is simply `id`.

Reply:
`Expr` field is the AST node that contains the aggregate function (i.e. `count(id)`), whereas `Original` is simply `id`.

Reply:
Hmm... Now am I'm even more confused. Why is the expression wrapped in AliasedExpr?

Reply:
Never mind. I get it now